central/authprovider/datastore/datastore_impl.go

```diff
@@ -73,33 +78,45 @@ func (b *datastoreImpl) RemoveAuthProvider(ctx context.Context, id string, force
 		return sac.ErrResourceAccessDenied
 	}
 
-	if err := b.verifyExistsAndMutable(ctx, id, force); err != nil {
+	ap, err := b.verifyExistsAndMutable(ctx, id, force)
+	if err != nil {
+		return err
+	}
+	if err = verifyAuthProviderOriginMatches(ctx, ap); err != nil {
 		return err
 	}
 	return b.storage.Delete(ctx, id)
 }
 
-func (b *datastoreImpl) verifyExistsAndMutable(ctx context.Context, id string, force bool) error {
+func verifyAuthProviderOriginMatches(ctx context.Context, ap *storage.AuthProvider) error {
+	if !declarativeconfig.IsOriginModifiable(ctx, ap.GetTraits().GetOrigin()) {
+		return errors.Wrapf(errox.InvalidArgs, "auth provider %q's origin is %s, cannot be modified or deleted within this context",
```

Comment:
Same notion within the other PR:
I would be in favor of `errox.NotAuthorized` + improved error message (i.e. dropping "within this context" and adding information regarding the permission).